src/Algebra/Graph.hs

```diff
@@ -232,49 +234,51 @@ instance Num a => Num (Graph a) where
     abs         = id
     negate      = id
 
+-- | `==` is a good consumer of both arguments.
 instance Ord a => Eq (Graph a) where
     (==) = eqR
 
+-- | 'compare' is a good consumer of both arguments.
 instance Ord a => Ord (Graph a) where
     compare = ordR
 
 -- TODO: Find a more efficient equality check.
 -- Check if two graphs are equal by converting them to their adjacency maps.
 eqR :: Ord a => Graph a -> Graph a -> Bool
 eqR x y = toAdjacencyMap x == toAdjacencyMap y
-{-# NOINLINE [1] eqR #-}
+{-# INLINE [2] eqR #-}
 {-# RULES "eqR/Int" eqR = eqIntR #-}
 
 -- Like 'eqR' but specialised for graphs with vertices of type 'Int'.
 eqIntR :: Graph Int -> Graph Int -> Bool
-eqIntR x y = toAdjacencyIntMap x == toAdjacencyIntMap y
+eqIntR = \x y -> toAdjacencyIntMap x == toAdjacencyIntMap y
```

Comment:
Is this change accidental? If not, what does it achieve?

Reply:
This was a little trick when I compared directly the definitions of functions and expected form. This is not needed anymore.